common/lessonplan/generate.ts

```diff
@@ -0,0 +1,298 @@
+
+    if (fileUuids.length > 0) {
+        // Fetch file contents for each UUID
+        const allFiles = await Promise.all(
+            fileUuids.map(async (uuid) => {
+                try {
+                    const file: File = getFile(uuid);
+                    const blob = new Blob([file.buffer], { type: file.mimetype });
+
+                    if (file.mimetype.startsWith('image/')) {
+                        // Handle image files
+                        logger.debug(`Processing image file: ${file.originalname}`);
+                        const resizedBuffer = await resizeImage(file.buffer);
+                        const base64 = resizedBuffer.toString('base64');
+                        const dataUrl = `data:${file.mimetype};base64,${base64}`;
+                        imageContents.push({
+                            type: 'image_url',
+                            image_url: {
+                                url: dataUrl,
+                            },
+                        });
+
+                        // // Generate a unique filename
+                        // const uniqueFilename = `${uuidv4()}${path.extname(file.originalname)}`;
+                        // const savePath = path.join('./', uniqueFilename);
+
+                        // // Write the resized image to a file
+                        // await fs.writeFile(savePath, resizedBuffer);
+
+                        logger.debug(`Image processed and converted to base64`);
+                        return null;
+                    }
+
+                    let loader;
+                    let docs;
+
+                    if (file.mimetype === 'application/pdf') {
+                        loader = new PDFLoader(blob, { splitPages: true });
+                        docs = await loader.load();
+                    } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.wordprocessingml.document') {
+                        loader = new DocxLoader(blob);
+                        docs = await loader.load();
+                    } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.presentationml.presentation') {
+                        loader = new PPTXLoader(blob);
+                        docs = await loader.load();
+                    } else {
+                        throw new Error(`Unsupported file type: ${file.mimetype}`);
+                    }
+
+                    // Add filename and page numbers (only for PDFs) to the content
+                    const content =
+                        `Filename: ${file.originalname}\n\n` +
+                        docs
+                            .map((doc, index) => (file.mimetype === 'application/pdf' ? `Page ${index + 1}:\n${doc.pageContent}` : doc.pageContent))
+                            .join('\n\n');
+
+                    // Check if the content is empty (excluding filename and page numbers)
+                    const contentWithoutMetadata = content
```

Comment:
`docs.some(doc => doc.length > 0)` - Reparsing the concatenated string looks a bit redundant?